recipes/auto_gptq/meta.yaml

```
@@ -0,0 +1,99 @@
+{% set name = "auto_gptq" %}
+{% set version = "0.5.1" %}
+
+{% set proc_type = "gpu" if cuda_compiler_version != "None" else "cpu" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  # url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/auto_gptq-{{ version }}.tar.gz
+  # sha256: a4897a59f8901f3c497c8d0a0422db40a79b23dbaf840fdb5de2fcf4482c8a54
+  url:  https://github.com/PanQiWei/AutoGPTQ/archive/v{{ version }}.tar.gz
+  sha256: 8a020bf61823261535229630445a5bbc53caaba03e491ee4f11ee38d59202203
+
+build:
+  # noarch: python
+  script_env:
+    - DISABLE_QIGEN=1 # coming soon
+  script: 
+    - |
+      CUDA_VERSION=$(/usr/local/cuda/bin/nvcc --version | sed -n 's/^.*release \([0-9]\+\.[0-9]\+\).*$/\1/p')
+      if [[ ${CUDA_VERSION} == 9.0* ]]; then
+          export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;7.0+PTX"
+      elif [[ ${CUDA_VERSION} == 9.2* ]]; then
+          export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0+PTX"
+      elif [[ ${CUDA_VERSION} == 10.* ]]; then
+          export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5+PTX"
+      elif [[ ${CUDA_VERSION} == 11.0* ]]; then
+          export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0+PTX"
+      elif [[ ${CUDA_VERSION} == 11.* ]]; then
+          export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6+PTX"
+      elif [[ ${CUDA_VERSION} == 12.* ]]; then
+          export TORCH_CUDA_ARCH_LIST="5.0;5.2;5.3;6.0;6.1;6.2;7.0;7.2;7.5;8.0;8.6;8.7;8.9;9.0+PTX"
+      else
+          echo "unsupported cuda version."
+          exit 1
+      fi
+      {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
```

Would recommend the following simplifications here. Please see [this example]( https://github.com/conda-forge/nvidia-apex-feedstock/blob/e90f97b8ce0a313b5bf76ac7b6d4ac414ff6bfff/recipe/meta.yaml#L15-L20 )

```suggestion
  script_env:
    - DISABLE_QIGEN=1 # coming soon
    - TORCH_CUDA_ARCH_LIST=3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6+PTX      # [cuda_compiler_version == "11.2"]
    - TORCH_CUDA_ARCH_LIST=3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6;8.9+PTX  # [cuda_compiler_version == "11.8"]
    - TORCH_CUDA_ARCH_LIST=5.0;6.0;6.1;7.0;7.5;8.0;8.6;8.9;9.0+PTX  # [(cuda_compiler_version or "").startswith("12")]
  script: {{ PYTHON }} -m pip install . -vv
```